rllib/algorithms/algorithm.py

```diff
@@ -1774,17 +1890,74 @@ def import_policy_model_from_h5(
 
     @override(Trainable)
     def save_checkpoint(self, checkpoint_dir: str) -> str:
-        checkpoint_path = os.path.join(
-            checkpoint_dir, "checkpoint-{}".format(self.iteration)
-        )
-        pickle.dump(self.__getstate__(), open(checkpoint_path, "wb"))
+        """Exports AIR Checkpoint to a local directory and returns its directory path.
+
+        The structure of an Algorithm checkpoint dir will be as follows::
+
+            policies/
+                pol_1/
+                    policy_state.pkl
+                pol_2/
+                    policy_state.pkl
+            checkpoint_version.txt
+            state.pkl
 
-        return checkpoint_path
+        Note: `checkpoint_version.txt` contains a version string (e.g. "v0") helping
+        RLlib to remain backward compatible wrt restoring from checkpoints from
+        Ray 2.0 onwards.
+
+        Args:
+            checkpoint_dir: The directory where the checkpoint files will be stored.
+
+        Returns:
+            The path to the created AIR Checkpoint directory.
+        """
+        state = self.__getstate__()
+
+        # Extract policy states from worker state (Policies get their own
+        # checkpoint sub-dirs).
+        # TODO: "state" key inside "worker" should be renamed to "policy_states".
+        policies = {}
+        if hasattr(self, "workers") and isinstance(self.workers, WorkerSet):
+            policies = state["worker"].pop("state", {})
+
+        # Add RLlib checkpoint version.
+        state["__version__"] = "v1"
+
+        # Write state (w/o policies) to disk.
+        state_file = os.path.join(checkpoint_dir, "state.pkl")
+        pickle.dump(state, open(state_file, "wb"))
+        # Write checkpoint version separately. This file will NOT be used
+        # by RLlib anywhere, it is solely for the user's convenience.
+        with open(os.path.join(checkpoint_dir, "checkpoint_version.txt"), "w") as f:
+            f.write(state["__version__"])
```

Comment:
can we write this metadata file in a separate util function?
I think it can be useful for us as well, I don't know if we want to claim this will NOT be used by RLlib ...
if we have the metadata dict, let's at least at timestamp, ray.__version__, and ray.__commit__ in there as well?